app/models/lookup_keys/lookup_key.rb

```diff
@@ -172,6 +142,10 @@ def overridden?(host)
     lookup_values.find_by_match(host.send(:lookup_value_match)).present?
   end
 
+  def puppet?
```

Comment:
seems like a massive overkill... this is needed because of legacy, no need for any other feature to be added at this time, why complicate this?

parent class can definitely define smart defaults, like in here, the default is that LK is not related to a puppetclass, only in instances of PuppetclassLK we will override this method to be true.